check_manifest.py

```diff
@@ -377,11 +378,24 @@ def get_vcs_files():
 
 
 def normalize_names(names):
+    """Normalize file names.
+    """
+    return list(map(normalize_name, names))
+
+
+def normalize_name(name):
     """Some VCS print directory names with trailing slashes.  Strip them.
 
     Easiest is to normalize the path.
+
+    And encodings may trip us up too, especially when comparing lists
+    of files.  Plus maybe lowercase versus uppercase.
     """
-    return [os.path.normpath(name) for name in names]
+    name = os.path.normpath(name)
+    name = os.path.normcase(name)
+    name = unicodify(name)
+    name = unicodedata.normalize('NFC', name)
```

Comment:
This is the sticking point: I feel uncomfortable doing this on Linux.

My signature file has had this quote for a long time:
> Please do not even think about automatically normalizing file names
> anywhere. There is absolutely no need for introducing such nonsense, and
> deviating from the POSIX requirement that filenames be opaque byte
> strings is a Bad Idea\[TM] (also known as NTFS).
>         -- Markus Kuhn

Unfortunately I can't find any rationale, or even the original source for the quote!

There's also the slight complication that OS X [doesn't actually use the Unicode NFD form, but a version of it](http://apple.stackexchange.com/a/84038).
E.g.

> In addition, the Mac OS X HFS+ file system considers certain Unicode codepoints as ignorable. For example, committing e.g. .g\u200cit/config and then pulling it on HFS+ would overwrite .git/config because U+200C is one of those ignorable codepoint. In general, Mac OS X normalizes filenames, and in many circumstances considers "different" filenames the same in many cases
--  http://www.dwheeler.com/essays/fixing-unix-linux-filenames.html

but

```
>>> unicodedata.normalize('NFD', u'g\u200cit/config')
u'g\u200cit/config'
```